### PR TITLE
wire the store with the API to fetch character list and characters

### PR DESCRIPTION
**Pull request by etnbrd**
This PR sets up the characters store, populated by the API client, and persisted across pages.

It also fixes various small details, such as:
- using union types rather than enum in the client
- adding `typecheck` command
- fixing reactivity by replacing all destructuring in the parameters with `props`

### CODE REVIEW

File: dnd5esheets/front/package.json [hunk 1]
```diff
@@ -5,12 +5,13 @@
     "build": "vite build",
     "start": "vite",
     "serve": "vite preview",
-    "generate-client": "openapi --input ./openapi.json --output ./src/5esheets-client",
+    "generate-client": "openapi --useUnionTypes --input ./openapi.json --output ./src/5esheets-client",
```
Comment (brouberol) [inline dnd5esheets/front/package.json:8]: Oh that's nice

File: dnd5esheets/front/package.json [hunk 2]
```diff
@@ -5,12 +5,13 @@
...
     "generate-effect-parser": "peggy --plugin ./node_modules/ts-pegjs/dist/tspegjs -o src/effects/parser/index.ts src/effects/parser/grammar.pegjs",
     "test": "vitest --run",
     "lint": "eslint src",
     "prettier": "prettier -w .",
-    "prettier-check": "prettier -c ."
+    "prettier-check": "prettier -c .",
+    "typecheck": "tsc --noEmit"
```
Comment (brouberol) [inline dnd5esheets/front/package.json:14]: Should we add this to `make front-check` ?

File: dnd5esheets/front/src/5esheets-client/models/AbilityName.ts [hunk 3]
```diff
@@ -3,11 +3,4 @@
...
 
-export enum AbilityName {
-    CHARISMA = 'charisma',
-    CONSTITUTION = 'constitution',
-    DEXTERITY = 'dexterity',
-    INTELLIGENCE = 'intelligence',
-    STRENGTH = 'strength',
-    WISDOM = 'wisdom',
-}
+export type AbilityName = 'charisma' | 'constitution' | 'dexterity' | 'intelligence' | 'strength' | 'wisdom';
```
Comment (brouberol) [inline dnd5esheets/front/src/5esheets-client/models/AbilityName.ts:6]: <3

File: dnd5esheets/front/package.json (hunk 2 above)
Comment (brouberol) [reply dnd5esheets/front/package.json:14]:
```
Found 23 errors in 3 files.

Errors  Files
    17  src/effects/index.ts:29
     5  src/effects/parser/index.test.ts:3
     1  src/effects/parser/index.ts:119
```

Maybe in another PR

**Review by brouberol — APPROVED** [review]
👏

File: dnd5esheets/front/package.json (hunk 2 above)
Comment (etnbrd) [reply dnd5esheets/front/package.json:14]: Precisely 😬 
I spent some time trying to type the effect parser, but couldn't get to a satisfying solution yet. I'll look at it again later.
In the man time, we could just ignore this file and add typecheck to `make front-check`.

**Pull request merged**